src/assignments/coded-messages.rst

```diff
@@ -0,0 +1,336 @@
+   b. Remember that functions can call other functions.
+   c. The ``decrypt`` function can be completed with only a small number of
+      statements.
+
+Test Your Functions!
+^^^^^^^^^^^^^^^^^^^^
+
+#. In ``main()`` define the ``message`` and ``shift_by`` variables. Assign
+   values to each one.
+#. Call the ``encrypt_with_shift`` function and use the two variables as the
+   arguments. Assign the returned (coded) message to another variable.
+#. Print the coded message.
+#. Call the ``decrypt`` function and send in the coded message and ``shift_by``
+   as the arguments. Assign the returned (translated) message to another
+   variable.
+#. Print the translated message.
+
+.. list-table:: Sample Results
+   :header-rows: 1
+
+   * - Message
+     - Function
+     - Shift
+     - Returned Message
+   * - LaunchCode
+     - ``encrypt_with_shift``
+     - 13
+     - YnhapuPbqr
+   * - Hello, World!
+     - ``encrypt_with_shift``
+     - 5
+     - Mjqqt, Btwqi!
+   * - Ktocji MJXFN!
+     - ``decrypt``
+     - 21
+     - Python ROCKS!
+   * - Hsz dljd T'x yze l nzopc? Hlens xp aczgp espx hczyr!
+     - ``decrypt``
+     - 11
+     - Who says I'm not a coder? Watch me prove them wrong!
+
+Part 3: Add User Input
+----------------------
+
+The next step is to make your program interactive.
+
+#. In ``main()``, code some ``input`` statements so the user can enter a
+   message and a shift value.
+#. Ask the user to choose to *encrypt* or *translate* the message.
+#. Print the original message and the result.
+
+Part 4: Create a Module
+-----------------------
+
+The ``alphabet_position`` and ``shift_character`` functions might be useful
+for other programs. Move them into a module!
+
+#. Create a new file in your project. Remember to add ``.py`` to the end of the
+   filename.
+#. Cut and paste the two functions into the new file. ``alphabet_position`` and
+   ``shift_character`` should no longer appear in the ``main.py`` file!
+#. Try running your program. It should NOT work! Read the error message
+   carefully.
+#. In ``main.py`` import the required function from the module, then run your
+   program again. Use any error messages to guide you as you get your code
+   working again.
+
+Sample Output
+-------------
+
+Your output does NOT have to look exactly like the samples, but it should be
+close.
+
+::
+
+   Enter a message: LaunchCode
+   Enter a shift value: 13
+   Would you like to [e]ncrypt or [t]ranslate the message? e
+
+   Original message: LaunchCode
+   New message: YnhapuPbqr
+
+::
+
+   Enter a message: Ktocji MJXFN!
+   Enter a shift value: 21
+   Would you like to [e]ncrypt or [t]ranslate the message? T
+
+   Original message: Ktocji MJXFN!
+   New message: Python ROCKS!
+
+Bonus Mission
+-------------
+
+The Caesar Cipher works fine, but it is VERY easy to decode. Add another
+function to your program that takes any coded message and returns all possible
+outcomes (one for each shift value).
```

Comment:
can you clarify this more? all possible outcomes for how the encrypted message can be translated back to an English message?

Reply:
I changed the text and added an example output.  Let me know what you think.